### PR TITLE
Bug fix 1421

### PR DESCRIPTION
**Pull request by OfirMos**
The issue:
https://github.com/RedisTimeSeries/RedisTimeSeries/issues/1421
The bug was fixed, by deleting the aggregation context and the last sample from the compaction key, and recreating the context from the older window.
All version where affected in some degree and from 1.8 it could cause a crash when trying to add new sample after the deletion when compaction is of type average.

### CODE REVIEW

File: src/compaction.c [hunk 1]
```diff
@@ -218,14 +220,17 @@ void AvgAddValue(void *contextPtr, double value, __attribute__((unused)) timesta
     }
 }
 
-void AvgFinalize(void *contextPtr, double *value) {
+int AvgFinalize(void *contextPtr, double *value) {
     AvgContext *context = (AvgContext *)contextPtr;
-    assert(context->cnt > 0);
+    _log_if(context->cnt == 0, "AvgFinalize: context->cnt is 0");
+    if (unlikely(context->cnt == 0))
+        return TSDB_ERROR;
```
Comment (filipecosta90) [inline src/compaction.c:227]: is it possible to add test for this scenario @OfirMos ?

File: src/chunk.c [hunk 2]
```diff
@@ -83,13 +83,15 @@ static Sample *ChunkGetSample(Chunk *chunk, int index) {
...
         RedisModule_Log(mr_staticCtx, "error", "Trying to get the last timestamp of empty chunk");
+        return 0;
     }
     return ChunkGetSample(chunk, ((Chunk *)chunk)->num_samples - 1)->timestamp;
 }
 
 double Uncompressed_GetLastValue(Chunk_t *chunk) {
     if (unlikely(((Chunk *)chunk)->num_samples == 0)) { // empty chunks are being removed
         RedisModule_Log(mr_staticCtx, "error", "Trying to get the last value of empty chunk");
+        return 0;
```
Comment (filipecosta90) [inline src/chunk.c:94]: is it possible to add test for this scenario @OfirMos ?

File: src/compaction.c [hunk 3]
```diff
@@ -443,34 +448,44 @@ static inline double variance(double sum, double sum_2, double count) {
...
 }
 
-void VarPopulationFinalize(void *contextPtr, double *value) {
+int VarPopulationFinalize(void *contextPtr, double *value) {
     StdContext *context = (StdContext *)contextPtr;
     uint64_t count = context->cnt;
-    assert(count > 0);
+    if (unlikely(count == 0)) {
+        *value = 0;
+        return TSDB_ERROR;
```
Comment (filipecosta90) [inline src/compaction.c:456]: is it possible to add test for this scenario @OfirMos ?

**Review by filipecosta90 — CHANGES_REQUESTED** [review]
LGTM, but if possible add tests to cover the 3 identified LOC.

File: src/chunk.c (hunk 2 above)
Comment (OfirMos) [reply src/chunk.c:94]: Nope, it's a dead code, meant to catch errors

**Review by filipecosta90 — APPROVED** [review]

**Pull request merged**